Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomUtils.cpp

```diff
@@ -64,6 +64,12 @@ namespace AzToolsFramework
                     settings.m_keepDefaults = true;
                 }
 
+                if ((flags & StoreInstanceFlags::StoreLinkIds) != StoreInstanceFlags::None)
+                {
+                    LinkIdMetadata linkIdMetadata;
```

Comment:
wouldn't this go out of scope immediately after this if block? The pointer to this object would point to garbage. Might work fine now since its an empty struct. Maybe this object should be scoped to the function instead of this if block?

Reply:
I was thinking the same thing, but we're doing the same thing with entityIdMapper right above?

Reply:
EntityIdMapper stays in scope until the load call is made. But LinkIdMetadata goes out of scope immediately after the if block executes.
@AMZN-koppersr  suggested an alternate option below, which when used will avoid scope issues like this altogether. I suggest going with that instead.